constants/chatbotCommands.py

```diff
@@ -809,8 +809,27 @@ async def tillerinoNp(fro: str, chan: str, message: list[str]) -> str | None:
         for _mods in match["mods"][1:].split(" "):
             mods_int |= mods.NP_MAPPING_TO_INTS[_mods]
 
-    # Get beatmap id from URL
-    beatmap_id = int(match["bid"])
+    if match["bid"] is not None:
+        # Get beatmap id from URL
+        beatmap_id = int(match["bid"])
+    else:
+        # They /np'ed a map, but no beatmap id is present.
+        # We'll need to use the set id to find a beatmap id.
+        beatmapset_id = int(match["sid"])
+        beatmap_rec = await glob.db.fetch(
+            """
+            SELECT beatmap_id
+            FROM beatmaps
+            WHERE beatmapset_id = %s
+            ORDER BY beatmap_id DESC
+            LIMIT 1
```

Comment:
Pretty arbitrary selection of beatmap id -- not much we can do here. Should be good enough I suppose.

Reply:
Could also opt to store the set id in the token.last_np, but it'd quite a bit more/deeper refactoring